ament_cmake_core/cmake/core/ament_add_default_options.cmake

```diff
@@ -0,0 +1,42 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Explicit call to add default CMake options (e.g. BUILD_SHARED_LIBS)
+#
+# .. note:: It can be called multiple times, but should be called
+#   once for any package that requires the options.
+#
+# :param EXCLUDE_BUILD_SHARED_LIBS: Exclude the BUILD_SHARED_LIBS option
+#
+# @public
+#
+macro(ament_add_default_options)
+  # TODO(CH3): Would be good to parse args to skip options next time.
+  set(options EXCLUDE_BUILD_SHARED_LIBS)
+  set(oneValueArgs)
+  set(multiValueArgs)
```

Comment:
Since CMake `macro()`s don't have a separate scope for variables, I'd recommend prefixing these variables to avoid collisions. Maybe `options` -> `_aado_options`?

Reply:
Ah good idea, I've been spoilt with how it's been used in the ignition stack :sweat_smile: 

https://github.com/ament/ament_cmake/pull/390/commits/cc27182977943951897255f0e12269032cdbc2eb